XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,99 @@
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways and therefore other apps may not know how to render the `conversationId` appropriately within the UI to reflect the corresponding brand. We propose a schema to standardize the use of conversation `metadata` for the purpose of client-side brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `brandInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: 'mydomain.xyz/abc/qrs',
+    metadata: {
+        brandInfo: {
+            displayName: 'My Company',
+            profileImage: 'mydomain.xyz/assets/myimage.png,
+            primaryColor: '#ffffff',
+        }
+    }
+}
+```
+
+Example `brandInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: 'galaxy.chat/dm/uniqueIdentifier',
+    metadata: {
+        brandInfo: {
+            displayName: 'Galaxy',
+            profileImage: 'galaxychat.com/assets/logo.png,
+            primaryColor: '#6865B8',
+        }
+    }
+}
+```
+
+The `profileImage` file is encouraged to meet the following criteria:
+
+- Aspect ratio: 1:1
+- Minimum resolution: 100x100
+- Maximum resolution: 800x800
+- Format: .PNG, .WEBP
+
+The `primaryColor` is a hex color code.
+
+## Backwards Compatibility
+
+Existing conversations will not be affected by the adoption of the new schema; only conversations created after the new schema is finalized will follow the updated schema.
+
+To maintain backwards compatibility, clients may consider rendering popular but older schemas (such as Lens), present in conversations created before the finalization of this XRC, in a manner consistent with their updated schema that follows this XRC.
+
+Old Lens schema:
+
+```json
+{
+        conversationId: lens.dev/dm/${memberA}-${memberB}
+        metadata: {}
+}
+```
+
+New Lens schema
+
+```json
+{
+    conversationId: 'lens.dev/dm/${memberA}-${memberB}',
+    metadata: {
+        brandInfo: {
+            displayName: 'Lens',
+            profileImage: 'lens.xyz/assets/myimage.png,
+            primaryColor: '#ffffff',
+        }
+    }
+}
+```
+
+## Security Considerations
+
+The metadata field can be spoofed by malicious apps to display Names and Images that degrade the user experience and are harmful to the brand’s perception. A mechanism for apps to sign messages and related metadata can enable front-ends to verify if the signature is from a credible source in order to prevent such spoofing. An affordance for client verifiability is under consideration.
```

Comment:
The invitation is encrypted and authenticated using the sender & recipient keys. So the metadata must have been created by one of the parties if the invitation is successfully unsealed (https://github.com/xmtp/proto/blob/main/proto/message_contents/invitation.proto).

Reply:
I see, I misread this section. This is however a more general concern that currently apps have complete access to a user's XMTP identity/keys, so a users must necessarily trust the app they are using in all regards not just this particular aspect. It may still be worth mentioning here, but I'm less sure about what the solution should be. Some amount of trust between the user and the app may always be necessary.